### PR TITLE
Run Dataflow for ML Engine summary in venv

### PR DESCRIPTION
**Pull request by mik-laj**
---
Issue link: WILL BE INSERTED BY [boring-cyborg](https://github.com/kaxil/boring-cyborg)

Make sure to mark the boxes below before creating PR: [x]

- [X] Description above provides context of the change
- [X] Unit tests coverage for changes (not needed for documentation changes)
- [X] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [X] Relevant documentation is updated including usage instructions.
- [X] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: airflow/providers/google/cloud/utils/mlengine_operator_utils.py [hunk 1]
```diff
@@ -225,20 +225,22 @@ def validate_err_and_count(summary):
...
+        py_file=os.path.join(os.path.dirname(__file__), 'mlengine_prediction_summary.py'),
         dataflow_default_options=dataflow_options,
         options={
             "prediction_path": prediction_path,
             "metric_fn_encoded": metric_fn_encoded,
             "metric_keys": ','.join(metric_keys)
         },
         py_interpreter=py_interpreter,
+        py_requirements=[
+            'apache-beam[gcp]>=2.14.0'
```
Comment (mik-laj) [inline airflow/providers/google/cloud/utils/mlengine_operator_utils.py:237]: We install these libraries in an isolated environment outside of user code. In this environment, we only run Airflow code. At this point, Airflow is not a framework, but an application. We do not need an extension point here, i.e. change versions. Version pinning is a good solution here to maintain stability.

File: airflow/providers/google/cloud/utils/mlengine_prediction_summary.py [hunk 2]
```diff
@@ -156,23 +157,22 @@ def run(argv=None):
...
+              | "Write" >> beam.io.WriteToText(
+                    prediction_summary_path,
+                    shard_name_template='',  # without trailing -NNNNN-of-NNNNN.
+                    coder=JsonCoder()
+            )
+        )
 
 
 if __name__ == "__main__":
+    logging.getLogger().setLevel(logging.INFO)
```
Comment (mik-laj) [inline airflow/providers/google/cloud/utils/mlengine_prediction_summary.py:179]: Dataflow does not print anything on the screen by default. Good practice says to configure the logger to be able to track the progress. This code is run in a separate process, so it's safe.

File: airflow/providers/google/cloud/utils/mlengine_operator_utils.py (hunk 1 above)
Comment (potiuk) [reply airflow/providers/google/cloud/utils/mlengine_operator_utils.py:237]: I am not sure if this is possible for example in Composer environment. But I like it.

Comment (mik-laj) [reply airflow/providers/google/cloud/utils/mlengine_operator_utils.py:237]: The Cloud Composer team must think of a way to access pip if they want this operator to work.  This team is following changes in MLEngine and is aware of new requirements.

Comment (mik-laj) [reply airflow/providers/google/cloud/utils/mlengine_operator_utils.py:237]: They can also prepare this virtual environment in a Docker image and provide it with the installation. In the next step, they can modify this call to use a pre-baked environment.

**Comment by mik-laj** [comment]
@potiuk Can I have your review again? Travis is happy.

**Review by potiuk — APPROVED** [review]
If Travis is happy, I am happy as well!

**Pull request merged**